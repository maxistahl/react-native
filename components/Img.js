import React, {useState} from 'react';
import {Image} from 'react-native';

export default (Img = ({src, style}) => {
  const [showDefault, setShowDefault] = useState(false);
  const icon = showDefault ? require('../assets/unknownuser.png') : {uri: src};

  return <Image style={style} source={icon} onError={() => setShowDefault(true)} />;
});
