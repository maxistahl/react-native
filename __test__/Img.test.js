/**
 * @jest-environment jsdom
 */
import React from 'react';
import {shallow} from 'enzyme';
import Img from '../components/Img';

describe('User Image', () => {
  describe('Rendering', () => {
    it('if the image uri provided is not available, render a default image', () => {
      const component = shallow(<Img src={'http://invalidURL/image.jpg'} />);
      expect(component).toMatchSnapshot();
    });
    it('if the image uri provided response a status 200, render the image as expected', () => {
      const component = shallow(<Img src={'https://tarmac.io/assets/images/testimonial-eric-johnson.png'} />);
      expect(component).toThrowErrorMatchingSnapshot();
    });
  });
});
