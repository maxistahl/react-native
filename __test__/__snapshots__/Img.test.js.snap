// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Image Rendering if the image uri provided is not available, render a default image 1`] = `ShallowWrapper {}`;

exports[`User Image Rendering if the image uri provided response a status 200, render the image as expected 1`] = `"received is not a function"`;
